plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditor.java

```diff
@@ -2747,9 +2747,9 @@ public boolean processQueries(@NotNull final List<SQLScriptElement> queries, fin
                     }
                 }
                 // Just create a new query processor
-                if (!foundSuitableTab) {
+                if (!foundSuitableTab && newTab) {
```

Comment:
We should create new processor only if it was explicitly requested (newTab == true)